tools/firtool/firtool.cpp

```diff
@@ -54,6 +54,11 @@ static cl::opt<bool> disableOptimization("disable-opt",
 static cl::opt<bool> lowerToRTL("lower-to-rtl",
                                 cl::desc("run the lower-to-rtl pass"));
 
+static cl::opt<bool>
+    enableLowerTypes("enable-lower-types",
+                     cl::desc("run the lower-types pass within lower-to-rtl"),
+                     cl::init(false));
```

Comment:
Please add a testcase that shows an example working with both modes (one that works, one that produces an error)

Reply:
@lattner is this what you had in mind? https://github.com/llvm/circt/pull/263/commits/0a64166528cf0b6bc82be3ac0089d1a4b531cdb9

In this PR, I fixed the issue Andrew brought up that led me to want to disable this by default. But this demonstrates the main issue that pass was created for--bundle types cannot be lowered to RTL without it.

Reply:
Looks great, thank you Mike!